diff --git a/lib/rules/no-relative.js b/lib/rules/no-relative.js
index 71594c83f1f4f733ffcc6047d7f7084348335dbe..d8623d87c89499c442171db3272cba07c9efabbe 100644
--- a/lib/rules/no-relative.js
+++ b/lib/rules/no-relative.js
@@ -41,7 +41,7 @@ module.exports = {
 			ImportDeclaration(node) {
 				const importPath = node.source.value;
 
-				if (!/^(\.?\.\/)/.test(importPath)) {
+				if (!/^(\.\.\/)/.test(importPath)) {
 					return;
 				}
 