diff --git a/lib/rules/no-relative.js b/lib/rules/no-relative.js
index a5f0f9d42b255077d10bacfe226ee2ddc0348991..6b7dfd90d2a2aeff77b115d2798f8a8080721d62 100644
--- a/lib/rules/no-relative.js
+++ b/lib/rules/no-relative.js
@@ -43,7 +43,7 @@ module.exports = {
       ImportExpression(node) {
         const importPath = node.source.value;
 
-        if (!/^(\.?\.\/)/.test(importPath)) {
+        if (!/^(\.\.\/)/.test(importPath)) {
           return;
         }
 
@@ -69,7 +69,7 @@ module.exports = {
       ImportDeclaration(node) {
         const importPath = node.source.value;
 
-        if (!/^(\.?\.\/)/.test(importPath)) {
+        if (!/^(\.\.\/)/.test(importPath)) {
           return;
         }
 
